### DISCUSSION TITLE
懒人包 2023V5

### DISCUSSION THREAD

**Discussion by hooke007** [hooke007/mpv_PlayKit#331] [回收]
- [x] <s>文档站未同步</s>
 **23.07.01** 已同步 [mpv-git-0bfafd2](https://github.com/mpv-player/mpv/commit/0bfafd2451b25d25d3a8249373dc63b5d5e94379)
- [x] 由于激进的采用了 `vulkan-copy` 作为首选的解码方式，如果发现存在解码错误，先前下载的使用者请从mpv.conf内 `--hwdec` 的列表中移除它
 **23.07.01** 已重发 标准包 修正
 当前为 `--hwdec=auto-copy,nvdec-copy` <s>注意 hwdec的回退机制实际仍是失效的 https://github.com/mpv-player/mpv/issues/11865</s> 可自行替换 `mpv.exe` (20230716之后的版本) 修复
- [x] 文件 `ETC_UAI_DML.vpy` 中存在错误，先前下载的使用者参考此提交进行手动修正 [`diff`](https://github.com/hooke007/MPV_lazy/commit/c2db14f04db4bfd467e97e4a2351363595a5c6e3#diff-3ec39de6f406296337099f73f7c6109ba8fa20f1ae79634221ea860e22e66a7c)
 **23.07.05** 已重发 vsLite vsMega 修正
- [x] yt-dlp 发布了cve修复（cookies泄露），先前下载的使用者可自行更新此组件 [`yt-dlp.exe`](https://github.com/yt-dlp/yt-dlp/releases/tag/2023.07.06)
 **23.07.07** 已重发 标准包 修正

<hr />
链接 <a href='https://github.com/hooke007/MPV_lazy/releases/tag/20230630'>mpv-lazy-20230630</a>

**Comment by ghost** [comment]


**Reply by ru251** [reply]
纯萌新疑问，请问除了标准包，其他的压缩包里都没有mpv.exe可执行文件，是不是要先安装mpv呢

**Reply by hooke007** [reply]
建议看完说明先

**Comment by daomengRen** [comment]
新手小白，请教下大佬，是不是标准包就可以满足一般的观影需求了，不用额外再去添加配置

**Reply by hooke007** [reply]
。。。。你自己的需求你自己都不了解我怎么知道呢
如果你的一般需求是打开视频只看，那是的。

**Reply by daomengRen** [reply]
好的，明白了，谢谢大佬回复

**Comment by 2607627233** [comment]
![捕](https://github.com/hooke007/MPV_lazy/assets/43110396/5fb2973f-be6f-49ce-8977-cc170c26127f)
UOSC自定义了一个控件 怎么把他变成图标啊a

**Reply by hooke007** [reply]
你看过知道怎么构建control之后，至少也把你写的发出来我才知道问题在哪

**Reply by 2607627233** [reply]
我就在Controls里面添加了这个
`command:置顶:cycle ontop?切换状态`

**Reply by hooke007** [reply]
`command:{icon}:{command}` 第二个参数是图标名，怎么会用中文呢

**Reply by 2607627233** [reply]
我搞不懂在哪定义这个icon,具体在哪个文件夹内？

**Reply by hooke007** [reply]
我的文档里不是写了吗，你再看看3.1.1 controls那一节，就一个网址

**Reply by 2607627233** [reply]
谢谢 解决了

**Comment by hyzmllcn** [comment]
mpv.exe打不开咋办 双击没反应

**Comment by 2607627233** [comment]
使用vulkan api，会在文件夹内生成log，但是打开是乱码
![QQ截图20230716185920](https://github.com/hooke007/MPV_lazy/assets/43110396/5b7ce20f-1b90-4733-9834-4423e2c50582)
参数
![QQ截图20230716185931](https://github.com/hooke007/MPV_lazy/assets/43110396/67c4cd61-7732-4cdc-ae45-27cad4afd1fc)
LOG
[mpv.20230716-185827-991.log](https://github.com/hooke007/MPV_lazy/files/12063828/mpv.20230716-185827-991.log)

**Reply by hooke007** [reply]
这不是mpv自己的日志。更像是系统的崩溃日志，自行更新播放器本体或用其他人的版本测试。依然存在那就向上游反馈

**Reply by 2607627233** [reply]
2周后才看到 解决了

**Reply by 2607627233** [reply]
又遇到一个新问题
![屏幕截图 2023-07-27 131733](https://github.com/hooke007/MPV_lazy/assets/43110396/12e71276-5ab1-4d45-9da4-f3118343c9e6)
![屏幕截图 2023-07-27 132535](https://github.com/hooke007/MPV_lazy/assets/43110396/1917091b-2e33-4bdb-9960-8742a05d9082)
![屏幕截图 2023-07-27 132556](https://github.com/hooke007/MPV_lazy/assets/43110396/2ed52acd-dba3-4e06-9670-6b48f43493ad)
是视频不支持vulkan吗，怎么能把这个命令行窗口去掉

**Comment by cancundeyingzi** [comment]


**Reply by cancundeyingzi** [reply]
![image](https://github.com/hooke007/MPV_lazy/assets/73635883/7839ea1c-2445-4278-a5ea-1be03b1fba3d)
去掉了俩#也闪退
#gpu-context = winvk后不闪退,难道是winvk的问题?

**Reply by cancundeyingzi** [reply]
win设置里一直是节能,昨天还正常,今天就寄了,也没改过配置

**Reply by cancundeyingzi** [reply]
> `VK_LOADER_LAYERS_DISABLE=~implicit~`

也是丢系统环境变量么

**Reply by cancundeyingzi** [reply]
b站视频闪退(mpv播放脚本)
本地视频/m3u直播源黑屏

**Reply by cancundeyingzi** [reply]
[mpv-lazy.log](https://github.com/hooke007/MPV_lazy/files/12092661/mpv-lazy.log)
hwdec-extra-frames=32似乎没有效果
日志里好像最后有说什么缓冲区

**Reply by cancundeyingzi** [reply]
我又重启了,现在正确调用vk,配置文件也没动,,见鬼,和最开始完全一样(我有备份,直接覆盖了)

**Reply by cancundeyingzi** [reply]
简单捣鼓了一下,hwdec = qsv-copy还是有很多问题,玄学毛病基本都是他干的,比如不支持杜比

**Comment by zfekin** [comment]
uosc的播放列表排序和windows的文件夹里按名称排序不一样。
例：当前播放的文件名为hao123.com-sc小狗，文件夹里下个文件是hao123.com-安全，但是播放器点下一个播放的是hao123.com-两个

**Reply by hooke007** [reply]
不是问题。uosc的排序是算法排的

**Reply by zfekin** [reply]
有什么方法能改修改吗

**Reply by hooke007** [reply]
win的排序算法不公开。
你想要使用系统条目的话就尝试一次性拖放多个文件进mpv（即创建播放列表

**Reply by zfekin** [reply]
好吧，我试了试mvp.net的排序不存在这种现象，和文件夹的排序是一样的

**Reply by hooke007** [reply]
mpvnet都不存在uosc文件管理这个功能，何来排序。。。

**Reply by zfekin** [reply]
不是，我意思是能不能把uosc的排序改成autoload脚本那样的，自动加载文件夹中的视频，和文件夹里的排序是一样的

**Reply by hooke007** [reply]
autoload的排序虽然改良过几次，但仍然做不到和系统一致，理由前面讲了。
至于uosc能不能进一步改良，你可以向uosc提出fr

（autoload可以和uosc一起使用）

你只要看的多了自然就会发现越来越多autoload和系统不一致的地方。此外系统的排序有时候也不合理（不如第三方文件管理器）。
<img width=500 src="https://github.com/hooke007/MPV_lazy/assets/41094733/1435ce12-0662-4381-9637-3119401e1f01">

所以要么不断改进uosc排序算法，要么用更好的文件浏览器，框选全部文件直接拖进去。
<img width=500 src="https://github.com/hooke007/MPV_lazy/assets/41094733/c6ff9f19-249b-457f-b7a9-63923f4dd82c">

**Comment by cancundeyingzi** [comment]
hwdec = qsv-copy和RTX 视频增强超分辨率似乎不兼容,
开了RTX 视频增强超分辨率,再启动mvp就会闪退.(只是似乎)
测试步骤:
1开机
2打开mpv,正常
3打开RTX 视频增强超分辨率4档
4浏览器视频播放,检测到显卡200w功耗,RTX 视频增强超分辨率正确开启
5打开mpv,窗口白屏,数秒后闪退

[mpv-lazy.log](https://github.com/hooke007/MPV_lazy/files/12123605/mpv-lazy.log)

**Reply by hooke007** [reply]
我没有条件复现这个。用no-config测向mpv报告吧

**Reply by hooke007** [reply]
https://github.com/mpv-player/mpv/issues/11981 你这报错方式有点乱。
除非能做到配置目录干净的和no-config一样，一般可能不接受带复杂配置的报错。
```
--no-config --hwdec=qsv-copy --vo=gpu-next --gpu-context=winvk
```
东西很多也很难确定是不是只是qsv的原因，涉及的要素太多

**Comment by chenydawn** [comment]


**Reply by hooke007** [reply]
net的内核版本早就停了，还是说你自己更新了版本？
试试自行更新/或者降级mpv吧

**Reply by chenydawn** [reply]
没有自己更新mpv版本 ，还是用的您发布的版本啥都没动。在懒人包中自行更新mpv 是直接替换mpv.exe 就行了吗？

**Reply by hooke007** [reply]
我的意思是mpvnet的liblacebo是远古版本，和现在mpv用的没有可比性。

是。

**Reply by chenydawn** [reply]
我用mpv-x86_64-v3-20230727-git-bc96b23  这个版本的 mpv 能正常打开
再把 mpv-x86_64-v3-20230727-git-bc96b23 里面的 mpv.exe 替换到 懒人包v4里面还是打不开。
我看的视频是emby 跳转到 mpv的。

**Reply by hooke007** [reply]
不要用emby测试，这对我不可控（也不止一次遇到报告用emby调用mpv时的莫名其妙的bug了），看直接用mpv-lazy是否存在问题。
如果不存在那基本是emby的问题

**Reply by chenydawn** [reply]
我有的时候看的是杜比视频所以才要用vo = gpu-next 这时候不会偏色  但是某些时候 举个例子emby调用mpv 完美世界第121集 4K HEVC  vo = gpu-next打不开，vo = gpu可以打开。完美世界第118集vo = gpu-next可以打开，vo = gpu可以打开。和emby公益服没有关系，其他emby服也是同样情况。
另一种情况 直接用emby看 不调用 mpv  视频流设置  gpu-next  可以正常打开。
主要是emby片源我没有办法下载下来，不了解视频规格。不用emby测试，不知道用什么片源规格测试。
gpu 也能用 ，只不过看杜比视频遇到偏色问题 就需要去 配置文件改成 gpu-next 。

**Reply by hooke007** [reply]
emby的问题我无法给予帮助。如前所述emby搭配mpv存在不少问题，你可以尝试去emby社区寻求答案。

**Reply by chenydawn** [reply]
十分感谢。

**Reply by chenydawn** [reply]
大佬，在播放器界面有快捷键或者选项让这两个值（vo = gpu，vo = gpu-next）来回切换吗？

**Reply by hooke007** [reply]
参考input。conf自行绑定

**Reply by chenydawn** [reply]
Ctrl+g            cycle-values vo "gpu-next" "gpu" 					 #! 切换gpu,gpu-next
 多谢大佬，这个指令播放器会重启一次生效

**Comment by horous** [comment]
大佬，我是萌新，有两个问题想请教下，第一个是在播放音乐时，能不能让界面有点儿东西，频谱，或者让那个图标能一直显示也是好的，不然纯黑的一片实在太难受了。我尝试改了配置文件，但是一直失败，如果能指点一下改哪里，万分感谢。
第二个是timeline已经播放过的地方能不能设置颜色呢？是修改哪个文件的哪里？
万分感谢

**Reply by hooke007** [reply]
不显示是因为没有vid 自己挂外部封面 https://mpv.io/manual/master/#options-cover-art-files
没有用户选项支持

**Reply by horous** [reply]
timeline的颜色自己没办法调整是不是？大佬能不能在后续的版本中加上这个选项啊？

**Reply by hooke007** [reply]
无此计划，我也不喜欢把uosc做成五彩斑斓的样子

**Reply by horous** [reply]
好吧，谢谢大佬啦。封面文件我在mpv.conf中加了两行，好像是行啦。感谢大佬，这个真是太好用啦！

**Reply by horous** [reply]
cover-art-file="D:\Program Portable\my-mpv\installer\mpv-icon.ico"
cover_art_file_persistance="audio"
大佬，我加了两行，结果音乐倒是出现了，但是播放视频的时候也成了这样图标，我觉得大佬的idle时候的界面特别好看，我想用在播放音乐的时候，但是不会设置。跪求指点

**Reply by hooke007** [reply]
第一个应该是个bug
cover_art_file_persistance 这个莫名其妙的代码哪里来的
idle界面也不是我做的，原版就是长这样，不支持播放时显示

**Reply by horous** [reply]
我在大佬给的链接中看到的要指定cover_art_file我就在mpv.conf中指定了，并参考大佬的样子，在下面加了persistance，然后，音乐的时候确实是显示了图标，但是视频时也成了播放图标了

**Reply by hooke007** [reply]
第一个参数问题追踪 https://github.com/mpv-player/mpv/issues/12219
不要创造不存在的参数，没有意义和作用

**Reply by horous** [reply]
大佬的意思是，cover-art-files = "D:\Program Portable\my-mpv\installer\mpv-icon.ico" 这行参数影响了视频的播放是个bug?

**Reply by hooke007** [reply]
追踪上述的链接

**Comment by 19184030** [comment]
我已经很认真的看了相关文档，但没有找到答案。请教
1. 如何设置单个进程即点即播（开多个视频的情况下只有一个播放窗口）？
2. 如何设置增加重新播放功能（单个视频播放结束后，点击播放或在现有界面前提下增加重播按钮以再次回放该视频）？
3. 如何调用系统存在的第三方视频音频codec播放它不识别的视频音频格式（例如Potplayer可以正确识别和播放的视频，DEMO参见网盘链接中的The Lord of the Rings Trilogy Trailer.flv）

链接：https://pan.baidu.com/s/1_5dCk0H90XiG4Y7e9HDajA?pwd=gh8b
提取码：gh8b

**Reply by hooke007** [reply]
1 不支持。已有重复帖子，自行搜索查看详情
2 命令 `seek 0 absolute ; set pause no`
3 不支持。mpv使用ffmpeg作为解码端

另外这个片子我粗看了一下编码很奇怪，虽然是hevc但是细节我不理解，不清楚是ffmpeg尚未实现id的识别还是单纯的编码错误导致的误判。
```
Could not find codec parameters for stream 1 (Video: none ([12][0][0][0] / 0x000C), none, 1971985 kb/s): unknown codec
```

**Reply by 19184030** [reply]
感谢解答。该视频编码解码的开发公司是北京视骏科技有限公司
参见http://strongene.com/cn/downloads/downloadCenter.jsp

**Comment by xuxucod** [comment]
请问一下，我用mpv明明是一倍速为什么感觉加速了好多倍，是什么原因造成的，头大，删了重下也没用

**Reply by hooke007** [reply]
一倍速是速度2还是1？

**Reply by xuxucod** [reply]
是速度1

**Reply by 2607627233** [reply]
n卡面板开垂直同步

**Reply by xuxucod** [reply]
感谢，问题已解决

**Comment by JsonBorn98** [comment]
V5版本的RIFE、MVT都无法对60帧的视频补帧吗，这是什么原因导致的？

**Reply by JsonBorn98** [reply]
新增了fps_max=61就可以了...

**Comment by Lightmarey** [comment]
mpv默认设置4倍速时静音，我查询到原因是scaletempo2 音频滤镜设置，但是mpvlazy中好像使用的音频滤镜不是这个，因此af=scaletempo2=max-speed 参数无效。
可以告知应该修改哪个参数嘛？

**Reply by hooke007** [reply]
你在哪里看到的用的不是这个

**Reply by Lightmarey** [reply]
不知道，但我觉得如果还是这个修改音频的话 修改mpv.conf应该有效。但添加该条后 4倍速时视频仍是静音状态，故有所推断？

**Reply by hooke007** [reply]
你写的是到底什么，你没有贴出来

**Reply by Lightmarey** [reply]
af=scaletempo2=max-speed=10

**Reply by zhongfly** [reply]
https://github.com/mpv-player/mpv/issues/12469
有人问了，https://github.com/mpv-player/mpv/pull/12479 这个pr好像是能改善

**Reply by Lightmarey** [reply]
谢谢，换用重编译过的版本并额外加入参数-audio-pitch-correction=scaletempo2后问题得到了解决。

**Reply by hooke007** [reply]
> `af=scaletempo2=max-speed=10`

这在我这里有效，无法复现

**Reply by hooke007** [reply]
该pr实际并没有真的和现有方案有区别，检查你的书写错误。因为上一个给我报告无效的用户证明是自己的编写错误 https://github.com/hooke007/MPV_lazy/discussions/310#discussioncomment-6273989

**Reply by Lightmarey** [reply]
并不，我处确乎必须手动指定audio-pitch-correction=scaletempo2参数才能使得af=scaletempo2=max-speed=10生效。

**Reply by hooke007** [reply]
无log的条件下没人知道发生了什么。且该pr已关闭，除非你自己维护一个版本解决这个问题

**Reply by Lightmarey** [reply]
好的，谢谢

**Comment by kaixc** [comment]
大佬救命
[hevcs]
profile-cond=get("video-format", "default_value") == "hevc"
vd-lavc-skiploopfilter=all
vf-add=rotate=PI/2 # 旋转90度

这个配置目的是让播放hevc视频的时候自动开启skiploopfilter，不然太卡了
问题是skiploopfilter这项配置好像不能在视频加载后修改，改了也没有生效，下边那个用来测试的旋转90度正常生效了

大佬救救，怎样让skiploopfilter这个生效，
不然每次播放hevc视频的时候，都要手动在配置文件里添加vd-lavc-skiploopfilter=all 这么一行，太麻烦了  

谢谢大佬

**Reply by hooke007** [reply]
先确定是否只对下一会话生效还是其它情况。

手动重载当前vid，如果不行就试试手动切文件，如果还不行就g。

**Reply by kaixc** [reply]
![image](https://github.com/hooke007/MPV_lazy/assets/34050027/d9697b27-cd3f-4c93-a1c6-1d22c57baf4f)
单独一个视频，没有下一话 大佬如何手动重载vid

**Reply by kaixc** [reply]
还有一条有用的线索，就是我发现控制台那行[auto_profiles] Applying tuto profile: hevcs 这行信息出现的顺序决定了配置会不会生效
比如在音视频加载后出现就不会生效比如这样
![image](https://github.com/hooke007/MPV_lazy/assets/34050027/b8fac877-70af-46b7-9806-0ed6b54ccb6a)

当我把判断那个==换成~=的时候，他出现在了第一行，这样就会生效
就像这样

![image](https://github.com/hooke007/MPV_lazy/assets/34050027/3f1944ff-a6a2-4034-b1ba-d9eae25c3f6f)

**Reply by hooke007** [reply]
不要问chatgpt这种具体软件问题，它毫无实际价值。

1. 你profile的类型写错了，这应该有自动回退参数你没写
2. 匹配的条件你也写错了

**Reply by hooke007** [reply]
重载vid就是先 `set vid no` 后 `set vid auto`

**Reply by kaixc** [reply]
可是控制台怎么显示已经应用了这个配置了，这不就说明那个判断为true了，下边旋转90度的也生效了，
为什么就vd-lavc-skiploopfilter=all这一项不生效，奇怪

**Reply by hooke007** [reply]
```
[test]
profile-cond = p.video_format=="hevc"
#profile-restore = copy
vd-lavc-skiploopfilter = all
```
可能需要重载vid

**Reply by hooke007** [reply]
> 可是控制台怎么显示已经应用了这个配置了，这不就说明那个判断为true了，下边旋转90度的也生效了， 为什么就vd-lavc-skiploopfilter=all这一项不生效，奇怪

不奇怪，说明需要vid初始化前启用才能有效，所以需要重载。旋转那是个滤镜，和 --vd这类参数完全不是一个管线上的，不能作为参考

**Reply by kaixc** [reply]
![image](https://github.com/hooke007/MPV_lazy/assets/34050027/0c0668cb-47d9-4f1e-9976-2d9b525edf62)
还是不行，这玩意儿只要出现在音视频下边就不生效，只有在第一行的时候会生效

**Reply by hooke007** [reply]
在控制台出现apply...之后，你手动重载了吗？

“出现在第一行才能生效”就是我说的在初始化之前，手动重载就是为了实现这个步骤。

**Reply by kaixc** [reply]
woc，大佬 果然重载后，生效了，
如何实现自动化，不然每次手动set vid no 然后 set vid auto
这样很麻烦不是吗

**Reply by hooke007** [reply]
只靠profile做不到。需要脚本

**Reply by kaixc** [reply]
多谢大佬指点 ，问题初步得到解决了，虽然有点瑕疵

![image](https://github.com/hooke007/MPV_lazy/assets/34050027/d41eb4f2-3f1c-4634-a253-c542a6539794)


这样配置可以实现自动化，就是能看到几毫秒的黑窗口，不过还可以接受

**Reply by hooke007** [reply]
你都写脚本了为什么要多此一举脚本里调用profile。。。直接脚本里运行命令改这个属性不就完事了

而且当前的问题是，这个属性在播放别的格式的时候并没有改回来。那不如直接写进mpv。conf作全局参数

**Reply by kaixc** [reply]
因为**vd-lavc-skiploopfilter=all**这个对与高画质的hevc视频画质几乎看不出影响，
但对于低画质画质影响比较大，所以不能放全局参数，
我的目的就是让mpv自动判断如果是hevc就开启这一项，反之就走默认的gpu-hq配置
因为电脑性能拉跨，高画质hevc不开这一项卡的一批，开了就流畅了

**Reply by hooke007** [reply]
你没有设定没有回退，激活之后已经是全局参数了

**Reply by kaixc** [reply]
woc翻遍文档终于找到最优解了

**Reply by kaixc** [reply]
![image](https://github.com/hooke007/MPV_lazy/assets/34050027/781f391e-9a72-4ce9-8117-de103eae4cde)
利用这个配置，也能达到视频重载的效果，而且速度更快几乎就是瞬间完成的，好家伙这下连那几毫秒的黑框都没了，而且还不用重头开始，而是直接跳到你播放的位置无缝对接，nice

**Reply by kaixc** [reply]
### .lua
```
function hevcGo()
    local format = mp.get_property("video-format")
    if format == "hevc" then
    
--      针对 HEVC 格式的视频，应用相关配置
        mp.set_property("profile", "hevcs")

--		利用切换 hwdec（硬件解码器）来重新初始化解码器
		mp.set_property("hwdec", "auto")
		mp.set_property("hwdec", "no")
		
--      在 OSD 上显示当前的 vd-lavc-skiploopfilter 设置
		mp.osd_message("vd-lavc-skiploopfilter ="..mp.get_property("vd-lavc-skiploopfilter"), 5)
	else
		mp.set_property("profile", "HQ")
    end
end
mp.register_event("file-loaded", function()
    hevcGo()
end)
```
### .conf
```
[HQ]
no-border								
profile=gpu-hq                          
profile=my_default                   

[hevcs]
vd-lavc-skiploopfilter=all
profile=my_default
osd-border-color="#fec6df"
```

**Reply by kaixc** [reply]
问题已解决，多谢大佬的帮助

**Reply by hooke007** [reply]
你根本不需要写这些profile啊，直接写进脚本啊，conf里写这些的意义是什么。。。

另外如果文件已经被判定为只能软解，你这么做是否能做到重新初始化解码器？

而且你仍然没有回退处理，激活之后仍然是全局参数

**Reply by kaixc** [reply]
这样做已经达到我想要的效果

当打开hevc视频
开启skiploopfilter
应用默认配置

当打开其他视频
应用默认配置加gpu-hq

**Reply by hooke007** [reply]
我的意思是，

1. 如果被判定只能软解，修改hwdec在我印象中是无效的。你用能硬解的视频是测不出来我说的这个问题的
2. 当前是hevc激活后，当你下一个视频不是hevc的时候，skiploopfilter并未执行关闭

额外，如果连续多个视频都是hevc，实际上并不需要做任何操作，当前会重复执行多余的操作。

**Reply by kaixc** [reply]
![594PT9WMKWI0KV4AHZO42{V](https://github.com/hooke007/MPV_lazy/assets/34050027/cb61752b-700c-41dc-b2ea-7c502f85f7d4)
打开非hevc视频是默认值，并不受影响，因为每次都是判断后，重新加载的配置文件，
而非hevc视频只加载了配置[HQ]里的内容，hevc也是只加载了[hevcs]里的内容，
经测试不会互相干扰

**Reply by kaixc** [reply]
> 额外，如果连续多个视频都是hevc，实际上并不需要做任何操作，当前会重复执行多余的操作。
目前只有这种办法了，

我也想过配置文件里把默认值写好自动加载，然后在脚本里切换那个属性，但是因为配置文件里默认已经开启了gpu-hq，我想在hevc视频的时候不开启gpu-hq , 问题是如何把profile=hpu-hq的内容，在运行时清空呢， 我试图用profile=default 结果清不掉，
所以索性 配置文件里默认都先不加载，用到那个加载那个，这样保证不会出错

**Reply by kaixc** [reply]
> 如果被判定只能软解，修改hwdec在我印象中是无效的。你用能硬解的视频是测不出来我说的这个问题的

### 目的是利用切换一下hwdec，初始化解码器从而让修改的配置生效，但要是碰到像你说的不能切换这个选项的视频，那就只能用重载vid了，但是重载vid性能比这个要差太多，体验很差
![image](https://github.com/hooke007/MPV_lazy/assets/34050027/6e95c5cc-d712-4dd9-8316-bd911d78ccce)

**Reply by hooke007** [reply]
> 当前是hevc激活后，当你下一个视频不是hevc的时候，skiploopfilter并未执行关闭

我不知道你是不是没展示完全，我测下来 vd-lavc-skiploopfilter 并没有如预期关闭。
(可能你把关闭写进my_profile了，但是我也没看到重载的操作，所以实际仍然是激活的，这靠检测参数是没用的)
你那边能用就行

**Reply by kaixc** [reply]
![image](https://github.com/hooke007/MPV_lazy/assets/34050027/e456f6a8-e8e6-44fb-8229-1b7d3595290b)

 ### 为了搞清楚激活后到底有没有影响下一次播放， 我直接把这里设置成只播放关键帧，这样就算不看参数也知道有没有生效
### 测试结果是，

- ### hevc视频只播放关键帧，卡顿

-  ### 其他视频正常播放，vd-lavc-skipframe 这项还是默认值

 **_另外除了检测参数，如何才能知道配置到底有没有激活 ？_**

**Reply by kaixc** [reply]
### 完整配置是这样的

- ### .conf
![image](https://github.com/hooke007/MPV_lazy/assets/34050027/500ed716-7f0e-4d4a-bfff-d6f738121895)

- ### .lua
![image](https://github.com/hooke007/MPV_lazy/assets/34050027/61878f07-cf47-45e0-ae87-eaaddc1b9109)

**Reply by hooke007** [reply]
我的之前结论仍然适用。
profile没有激活的概念，它是参数的集合。应用一个profile等效 逐条应用集合内的参数。而且profile可以无限嵌套和无限叠加应用

**Reply by kaixc** [reply]
### 其实理想的方案是，配置文件默认加载[A]配置和[B]配置，
### 当hevc视频的时候,保留[A]配置的内容，把[B]配置的内容清空，同时加上skiploopfilter
### 当非hevc的时候什么都不用做，相当于只做了一个判断，反正每次打开视频都要判断是必不可少的

**Comment by signin404** [comment]
((p.file_size/1024)*8)/p.duration<=3000
这个在vpy里要怎么写?

我这样写显示未定义
asd = file_size
ddd = duration
if ((asd/1024)*8)/ddd<=3000 :

**Reply by hooke007** [reply]
vs不支持mpv的属性

**Comment by signin404** [comment]
播放列表的文件名变成标题的功能有开关么?

**Reply by hooke007** [reply]
自行查阅脚本设置文件，有就是支持，没有就是不支持

**Comment by signin404** [comment]
> 自行查阅脚本设置文件，有就是支持，没有就是不支持

这是哪里写错了?

![QQ截图20231007130331](https://github.com/hooke007/MPV_lazy/assets/145902543/7fe856e7-1ea6-4157-818d-9dc25dcc5107)

**Reply by hooke007** [reply]
这都不是同一个脚本

**Comment by michael155183** [comment]
萌新請問已經有設定SVP為vf濾鏡了但為何沒有補幀效果?  PS:如果問了傻問題請大佬見諒(´～`)
![螢幕擷取畫面 2023-10-08 100012](https://github.com/hooke007/MPV_lazy/assets/79831851/19da4f02-cc98-4141-87c5-8241865d9cb3)

**Reply by hooke007** [reply]
看控制台

**Reply by michael155183** [reply]
打開控制台之後跳出No attribute with the name svp1 exists錯誤訊息，我有嘗試去搜尋上面的檔案去想去修改但都找不到，請問大佬有解嗎?
![MPV-00_01_58 852-N0001](https://github.com/hooke007/MPV_lazy/assets/79831851/560972fe-996f-4e81-b11d-6ac9586cf87f)

**Reply by hooke007** [reply]
说明vs没找到运行所需的对应组件。按照说明正确安装滤镜所需组件

**Reply by michael155183** [reply]
我後又研究了一下原來SVP組件沒在vsMega包裡面，把網上找的svp1組件丟到vapoursynth64\plugins資料夾裡就可以了，感謝大佬願意解答我的傻問題m(＿＿)m(膜拜)

**Comment by kaixc** [comment]
### 加载在线视频问题

- 最新安装的纯净标准版
![image](https://github.com/hooke007/MPV_lazy/assets/34050027/154ef77d-fab0-4afd-9c14-f6a1ece6c20b)

- 按照这里的讨论，添加了相关配置https://github.com/hooke007/MPV_lazy/discussions/308#discussion-5181774
- 问题是：可以成功解析出高分辨率视频，但是播放不出来
- 加载方式：
![image](https://github.com/hooke007/MPV_lazy/assets/34050027/fb3cd329-25e3-4f34-a396-575b03de0ba6)

- 一直报这个错误
![image](https://github.com/hooke007/MPV_lazy/assets/34050027/fe6ecfb4-ae91-4ee6-8750-e1dde9e862f8)

**Reply by hooke007** [reply]
报错来自ffmpeg，超纲了，大概率和mpv本身无关
我只能建议单独更新ytdlp和mpv再次测试

**Reply by kaixc** [reply]
### 大佬，已找到原因

- ### **在最新的懒人包【[mpv-lazy-20230630.exe](https://github.com/hooke007/MPV_lazy/releases/download/20230630/mpv-lazy-20230630.exe)】里**
- ![image](https://github.com/hooke007/MPV_lazy/assets/34050027/886afec6-0654-4165-be3d-c6d9acd964e1)
### 这个[yt-dlp.exe]版本由于获取不到弹幕，从而一直报错而且非常卡
- ### 换成这个版本问题解决 https://github.com/hooke007/MPV_lazy/discussions/308#discussioncomment-6325198
- ### 不用登陆账号，直接复制链接，loadfile就可以在线播最高画质，还有弹幕
- **建议替换这个文件**

**Reply by hooke007** [reply]
有流媒体需求建议自行更新该文件。这是由于平台频繁更新的特性决定的，我不可能追踪每一次更新

**Reply by kaixc** [reply]
>https://github.com/hooke007/MPV_lazy/discussions/308#discussion-5181774 我想探讨的是把cookies存于本地,调用yt-dlp.exe解析链接的模式,并不需要自己的账号实际点进链接的模式

**由于我没有会员账号，用这种方法可以完美的解决在线看高清的需求，很不错的功能**

**Comment by signin404** [comment]
alang和slang写有相同的值会导致字幕自动加载失效是BUG么?

alang=jpn
slang=chs,sc,zh,chi,zho,cht,tc,jpn
播放仅有jpn音轨的视频外挂字幕不会自动加载
把slang的jpn删除就正常

**Reply by hooke007** [reply]
检查你版本所对应的参数 https://mpv.io/manual/master/#options-subs-with-matching-audio

**Reply by signin404** [reply]
yes有效 刚才测错了
不过这不是默认值么?

**Reply by hooke007** [reply]
“检查你版本所对应的参数”
重点是你的版本而不是网页的版本

**Comment by 2607627233** [comment]
![image](https://github.com/hooke007/MPV_lazy/assets/43110396/38e36891-7b1a-4f9b-a163-4c49aca370e9)
上面hermite是设置的参数
下面 bicubic 在更新了mpv版本之后播放HDR视频就会出现，是执行了两次吗，还是什么新的显示方式

**Reply by hooke007** [reply]
这个难道不是在处理hdr相关的stage？ https://github.com/mpv-player/mpv/issues/11841#issuecomment-1611185391
